### DISCUSSION TITLE
How to change the default style of DataGridComboBoxColumn?

### DISCUSSION THREAD

**Discussion by Diyari-Kurdi** [MaterialDesignInXAML/MaterialDesignInXamlToolkit#3019] [Q&A] [answered]
### How can i style a **DataGridComboBoxColumn** with **AutoGenerateColumns** set to **True**?

```
<DataGrid
          AutoGenerateColumns="True"
          ItemsSource="{Binding Items}"/>
```

![MaterialDesignDataGrid](https://user-images.githubusercontent.com/65598554/210111422-ae682247-e7e7-4ac5-92ee-71c94db3a7d2.png)

**Comment by nicolaihenriksen** [comment, marked as answer]
@Diyari-Kurdi When you let the `DataGrid` auto-generate columns rather than explicitly defining them in XAML, you're effectively cutting specialized MDIX columns out of the loop. However, there is still a hook you can use to obtain what I believe you're going for.

My assumption from your sample is that the underlying data source for the "Continents" column is an `enum`. If so, you can attach the hook to your data grid like this:

```xaml
<DataGrid AutoGenerateColumns="True"
          AutoGeneratingColumn="DataGrid_OnAutoGeneratingColumn"
          ... />
```

Then in your code-behind you can implement the hook in a manner like this:

```csharp
private void DataGrid_OnAutoGeneratingColumn(object sender, DataGridAutoGeneratingColumnEventArgs e)
{
    if (e.PropertyType.IsEnum)
    {
        e.Column = new MaterialDesignThemes.Wpf.DataGridComboBoxColumn { Header = e.Column.Header, ItemsSource = Enum.GetValues(e.PropertyType) };
    }
}
```

There may still be some things with data binding and such that you need to iron out. This is just ensuring that you visually get the MDIX style for the `ComboBox`.

<img width="627" alt="image" src="https://user-images.githubusercontent.com/19572699/210244586-32fd8c12-a2d1-48b0-a442-5d68fc04fedc.png">

For more information on the topic, you can go here:
https://learn.microsoft.com/en-us/windows/communitytoolkit/controls/datagrid_guidance/customize_autogenerated_columns

**Reply by Diyari-Kurdi** [reply]
@nicolaihenriksen Thank you sir.

**Reply by nicolaihenriksen** [reply]
@Keboo The library does setup default styles for some auto-generated columns (`CheckBox` and `Text` - see [here](https://github.com/MaterialDesignInXAML/MaterialDesignInXamlToolkit/blob/f075b3091b642981589d088033873b3a5835db31/MaterialDesignThemes.Wpf/Themes/MaterialDesignTheme.DataGrid.xaml#L552-L555)), it may be worth considering if other column types should be added to this (like for example the `ComboBox` column)?

**Reply by Keboo** [reply]
Yep agreed, I like what you are doing in your draft PR so we can work from there.
